tasks/main.yml

```diff
@@ -1,28 +1,45 @@
 ---
+# Reboot if required before installing new updates
+- name: Check if reboot is needed
+  ansible.windows.win_powershell:
+    script: |
+      $RebootPending = Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending"
+      $RebootRequired = $RebootPending `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired") `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootPending")
+      if ($RebootRequired) {
+          Write-Output "YES"
+      } else {
+          Write-Output "NO"
+      }
+  register: reboot_needed
 
-#Reboot if required before installing new updates
-- name: If reboot required before checking and installing new updates, reboot the host.
-  win_reboot:
-   reboot_timeout_sec: 1500
-  when: ansible_reboot_pending
+- name: Reboot host before check for update
+  ansible.windows.win_reboot:
+    reboot_timeout_sec: 3600
```

Comment:
Is this really long timeout necessary? Waiting 1 hour seems really odd and something should be investigated. Using the default value (600) seems really reasonable, as per the docs: This timeout is evaluated separately for both reboot verification and test command success so maximum clock time is actually twice this value. ([ref](https://docs.ansible.com/ansible/latest/collections/ansible/windows/win_reboot_module.html#parameter-reboot_timeout))

Reply:
I understand that for specific cases a machine may be slow and have lots of updates to apply, hence you may need to wait a long time. Possibly ignore this comment then

Reply:
Considering that some reboots take a long time in the update process, I consider it more prudent to change to 30 min. The interesting thing about this module is that if the reboot takes only 10 min, it will not wait 30 min, but only 10 min.